client/db/bolt/db_test.go

```diff
@@ -224,6 +224,70 @@ func TestAccounts(t *testing.T) {
 	}
 }
 
+func TestDeleteAccount(t *testing.T) {
+	boltdb := newTestDB(t)
+	acct := dbtest.RandomAccountInfo()
+	host := acct.Host
+	boltdb.CreateAccount(acct)
+
+	err := boltdb.deleteAccount(acct)
+
+	if err != nil {
+		t.Fatalf("Unexpected deleteAccount error: %v", err)
+	}
+	actualAccount, err := boltdb.Account(host)
+	if err == nil {
+		t.Fatalf("Expected account not found error")
+	}
+	if actualAccount != nil {
+		t.Fatalf("Did not expect to retrieve deleted account.")
+	}
+}
+
+func TestDisableAccount(t *testing.T) {
+
+	boltdb := newTestDB(t)
+	acct := dbtest.RandomAccountInfo()
+	host := acct.Host
+	boltdb.CreateAccount(acct)
+
+	err := boltdb.DisableAccount(acct)
+
+	if err != nil {
+		t.Fatalf("Unexpected DisableAccount error: %v", err)
+	}
+	actualAcct, _ := boltdb.Account(host)
+	if actualAcct != nil {
+		t.Fatalf("Expected retrieval of deleted account to be nil")
+	}
+	actualDisabledAccount, err := boltdb.DisabledAccount(acct.EncKey)
+	if err != nil {
+		t.Fatalf("Unexpected DisabledAccount error: %v", err)
+	}
+	if actualDisabledAccount == nil {
+		t.Fatalf("Expected to retrieve a disabledAccount.")
+	}
+}
+
+func TestDisabledAccount(t *testing.T) {
```

Comment:
This test is already covered by `TestDisableAccount`. Can delete, I think.

Reply:
I also had meant to make this comment. For some reason it's not showing up though...
These tests are already covered.

Reply:
There is extreme overlap between these two tests and the underlying function names are so similar that it's weird and probably confusing.

That being said I was creating an explicit test for each function. 

`TestDisableAccount()` ensures that `DisableAccount()` removes the account from the accounts bucket and places the disabled account into the disabledAccounts bucket.

`TestDisabledAccount()` ensures that `disabledAccount()` is able to retrieve records from the disabledAccounts bucket.

This way we could theoretically change the behavior of `DisableAccount()` without affecting the test result for `disabledAccount()`. 

I just wanted to provide insight to my thoughts on this. It looks unanimous that focus on code coverage is preferred so I will remove TestDisabledAccount().

Reply:
> This way we could theoretically change the behavior of DisableAccount() without affecting the test result for disabledAccount().

My two cents:
Which would be good, but the tests are the same. The test for `disabledAccount()` uses `DisableAccount()`. I also think the probably the "correct" answer would to be to write a separate `disableAccount()` function just for tests, but then it also could be seen as a waste of time, as the function would in the end be exactly like `DisableAccount()`. I think it's best to compromise the test "rules" here for the sake of DRY (don't repeat yourself) which decreases time spent now, upkeep costs, and confusion of having multiple functions doing the same thing.

Again just my opinion. Worth a good conversation tho.

Reply:
Splitting hairs here but there was actually one difference.  The `DisableAccount()` test checked if the account was deleted by attempting to retrieve the account while the `disabledAccount()` test did not, it only cared if if retrieved the disabled account.

I did consider creating a `CreateDisabledAccount()` function which would strictly be used by the `disabledAccount()` test but I thought that would just be going further down the rabbit hole of creating functions to satisfy testing.